### DISCUSSION TITLE
Image screenshot saved?

### DISCUSSION THREAD

**Discussion by louis-vgn** [dynobo/normcap#332] [Q&A] [answered]
Does Normcap actually takes a screenshot (an image screenshot). If so, does it store it somewhere? If so, where is it stored?

**Comment by dynobo** [comment, marked as answer]
Hi @louis-vgn ,  NormCap itself normally doesn't store the images of the screenshots it takes. But to be precise, there are two cases in which images end up on your harddrive anyway:

1. If you [run NormCap in debug mode](https://dynobo.github.io/normcap/#faqs-debug) (e.g. `normcap -v debug` ) some images are written to a temporary folder to support troubleshooting. On Linux, the images should be written to `/tmp/normcap/`, but the log output contains the complete images' paths, e.g.:

    ```
    11:05:45 - DEBUG   - normcap.gui.utils:37 - Store debug image in: /tmp/normcap/2022-12-24_11-05-45_764832_enhanced.png`
    ```

2. If you run a Linux system with the Wayland compositor (according to #331 you do ;-), NormCap has to use Wayland's internal screenshot mechanism, which often (or always?) stores the full screenshot somewhere in your home directory.  For me, they end up in `~/pictures/`. 
    (I considered trying to detect and delete those images automatically, but it is rather complex, and also it didn't feel right to me to mess with the users' files.)

Does this answer your question?

**Reply by louis-vgn** [reply]
Yes, ok, I see. However, on my computer it doesn't seem to store the screenshots in `~/Pictures`... I'd have to investigate a bit more :D.

Also, maybe you thought about it already, but while messing with User's files and deleting them doesn't feel right, placing the screenshots images in `/tmp/normcap/` may be an idea? Screenshots are not deleted directly, so we can still access it, but they don't use space once rebooting.

Anyway, thanks again.

**Reply by dynobo** [reply]
Hi @louis-vgn, 

do you mind elaborating a bit on your use-case for saving the images? In what kind of situations would this be helpful to you? Are you interested in the images of the full screen or just the selected portion?

If it turns out to be useful for more users, I might add it as an optional feature...

**Reply by louis-vgn** [reply]
For me, there is no reason to save the image; it’s more annoying than anything else. If I want a screenshot, not a Normcap, I actually take a screenshot. If I want a Normcap, I take a Normcap, and I only want the text to be in my clipboard. However, since you said you didn’t want to mess with the user’s files, which I understand, why not put the screenshots in `/tmp/normcap/`, so that if some users want to retrieve the screenshots images they can, while others who don’t want (like myself), we don’t have to worry about screenshots images polluting the file system?

**Reply by dynobo** [reply]
Ah, I see. Unfortunately, I can neither control _if_ nor _where_ gnome/wayland is storing those screenshots...
